### PR TITLE
Refactor composable filter

### PR DESCRIPTION
**Pull request by anitavincent**
#### What is the purpose of this pull request?
Start migrating filter API to composable

#### Next steps / TODO

- Rename value variable
- Change label prop from 'children' to 'label'
- Add csx compatibility
- Change style of footer to use stack if possible
- Improve typing differentiation between states
- Remove the need for state on every component with context
- Make id on every option optional (see what i did there? ( ͡° ͜ʖ ͡°) )
- Update docs
- More stories
- Remove usesystem from listbox

#### How should this be manually tested?
Check Filters section on storybook

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Quality improvement (tests or refactors)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Trivial change (small fix or feature that doesn't impact functionalities)
- [ ] Requires change to documentation, which has been updated accordingly

### CODE REVIEW

File: packages/admin-ui/src/filters/__stories__/filter.stories.tsx [hunk 1]
```diff
@@ -165,8 +260,82 @@ export function Interationalization() {
...
+        <FilterListbox state={filterState.combobox}>
+          {items.map((item) => (
+            <FilterOptionRadio item={item} key={item.id} state={filterState} />
+          ))}
+        </FilterListbox>
+        <FilterFooter {...filterState} />
+      </FilterPopover>
+    </>
+  )
+}
```
Comment (matheusps) [inline packages/admin-ui/src/filters/__stories__/filter.stories.tsx:341]:
```suggestion
export function SingleFromScrach() {
  const filter = useFilterState()

  return (
    <>
      <FilterDisclosure state={filter}>Example</FilterDisclosure>
      <FilterPopover state={filter}>
        <FilterListbox state={filter}>
          <FilterOptionRadio>Full</FilterOptionRadio>
          <FilterOptionRadio>Empty</FilterOptionRadio>
          <FilterOptionRadio>Half full</FilterOptionRadio>
        </FilterListbox>
        <FilterFooter state={filter} />
      </FilterPopover>
    </>
  )
}
```

File: packages/admin-ui/src/filters/__stories__/filter.stories.tsx [hunk 2]
```diff
@@ -1,14 +1,19 @@
 import React, { useEffect } from 'react'
```
Comment (anitavincent) [inline packages/admin-ui/src/filters/__stories__/filter.stories.tsx:1]: This file changed a lot, it's easier to see the file by itself (without diffing)

File: packages/admin-ui/src/filters/filter-footer.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+      },
+    },
+    children: (
+      <>
+        <Button variant="tertiary" onClick={onClear}>
+          {formatMessage('clear')}
+        </Button>
+        <Button onClick={onChange}>{formatMessage('apply')}</Button>
+      </>
+    ),
```
[the remark this answers is not in the document]
Comment (anitavincent) [reply packages/admin-ui/src/filters/filter-footer.tsx:39]: I'll check it out later as we talked

**Review by lucasaarcoverde — APPROVED** [review]
LGTM

**Pull request merged**